compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/UtilLF.hs

```diff
@@ -109,3 +109,21 @@ instance Outputable Expr where
 sourceLocToRange :: SourceLoc -> Range
 sourceLocToRange (SourceLoc _ slin scol elin ecol) =
   Range (Position slin scol) (Position elin ecol)
+
+mkBuiltinEqual :: Version -> BuiltinType -> Expr
+mkBuiltinEqual v ty =
```

Comment:
Should we use these helpers in `DA.Daml.LFConversion.Primitives` as well?

Reply:
Can't, those are more general (they work for any type, not just builtins).

Reply:
Sorry should have been more clear: Would it make sense to generalize them so they work for both?

Reply:
I think it's a bit of a mismatch, it doesn't really work. It may look like there's duplication here, but there really isn't. 

The reason is the implementation here is just replacing the case where you want to compare a builtin type, because that's what is needed in and that's all that is available for older LF versions. OTOH, the implementation in Primitives.hs is exposing new functionality (comparison of types that are not builtin) when it is available. If I really wanted to unify these I would have to introduce a compiler error for the case where this function is called on a non-builtin type in older LF versions.

Reply:
Oh I see, thanks for the explanation!

Reply:
it's times like these that a very small amount of dependent types would help . . .